### PR TITLE
Add some PBC overview and PBC SCF section to user guide

### PR DESCRIPTION
**Pull request by tberkel**

### CODE REVIEW

File: source/user/pbc.rst [hunk 1]
```diff
@@ -1,13 +1,36 @@
...
+PBC-specific functionality must be imported from :mod:`pyscf.pbc`,
+which has a directory structure that mirrors that of the molecular :mod:`pyscf`
+module, e.g. ``from pyscf.pbc import gto, scf``.
+Details of how to specify the system details, including the unit cell,
+basis sets, and pseudopotentials, are provided in
+:ref:`user_pbc_gto`.
 
+All electronic structure methods can be applied with periodic
+boundary conditions at the Gamma point of the Brillouin zone, :math:`k=(0,0,0)`,
+as described in :ref:`mix_mol`.  Converging to the thermodynamic limit
```
Comment (gkc1000) [inline source/user/pbc.rst:18]: Should we say that we can use molecular methods at any point of the Brillouin zone? Although only k=(0,0,0) corresponds to PBC. If we have a twist averaging example in the examples directory, could link to that.

**Review by gkc1000 — APPROVED** [review]

**Comment by tberkel** [comment]
My fear is that non-Gamma-point requires complex support, which I'm not
sure all molecular methods have.

**Comment by gkc1000** [comment]
I think almost all methods do. But we should probably have a list
somewhere. I'll put that on the todo on slack.

**Comment by sunqm** [comment]
complex values were explicitly considered in mp2, cisd, ccsd only

File: source/user/pbc.rst [hunk 2]
```diff
@@ -1,13 +1,36 @@
...
+requires the use of larger and larger supercells, which might quickly become
```
Comment (susilehtola) [inline source/user/pbc.rst:19]: which may quickly become

File: source/user/pbc/scf.rst [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+calculated non-self-consistently along a k-point path using the
+``SCF.get_bands(kpts)`` function, where ``kpts`` is a list of k-points along
+which the band structure is desired.
+
+.. warning::
+   For Hartree-Fock or hybrid DFT, the discontinuity of the exchange potential
+   at :math:`G=0` is problematic for band structure calculations.  Using
+   :attr:`exxdiv='vcut_sph'` with FFTDF is recommended instead.  Alternatively,
+   the SCF procedure can be repeated at each k-point, which is much more
+   expensive but allows the use of any :attr:`exxdiv` or density fitting.
```
Comment (susilehtola) [inline source/user/pbc/scf.rst:128]: density fitting -> density fitting scheme

File: source/user/pbc/scf.rst [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+All molecular SCF add-ons are also available for periodic SCF but
+must be accessed through the molecular :mod:`pyscf.scf.addons` module.  
+Here we highlight a few of the most useful add-ons.
+
+Linear dependencies
+-------------------
+
+The dense packing of atoms in solids combined with the use of diffuse
+atom-centered basis functions is responsible for frequent linear dependencies.
+The linear dependency can be eliminated by Cholesky orthogonalization::
```
Comment (susilehtola) [inline source/user/pbc/scf.rst:145]: I think it still uses canonical by default

File: source/user/pbc/scf.rst [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    kmf = mol_scf.addons.remove_linear_dep_(kmf).run()
+
+Smearing
+--------
+
+For metals or small band gap semiconductors, it can be useful to smear the
+orbital occupation numbers away from integer values.  This can improve SCF
+convergence and can expedite convergence to the thermodynamic limit with k-point
+sampling.  Because this approach assumes a finite electronic temperature, it
+yields an entropy and free energy::
```
Comment (susilehtola) [inline source/user/pbc/scf.rst:158]: and a free energy

File: source/user/pbc/scf.rst [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+   Because most functions in PySCF assume integer occupations, they may fail if
+   combined with a mean-field calculation that was performed with smearing.
+
+Stability
+---------
+
+Periodic SCF solutions can be checked with stability analysis::
+
+    kmf = scf.KRHF(cell).run()
+    kmf.stability()
```
Comment (susilehtola) [inline source/user/pbc/scf.rst:179]: Is smearing supported in combination of stability analysis?

Comment (sunqm) [reply source/user/pbc/scf.rst:179]: Not supported

**Pull request merged**